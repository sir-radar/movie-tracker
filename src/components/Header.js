import React from 'react';

const Header = () => {
  return (
    <header className="row bg-header p-3 text-light font-weight-bolder">
      Movie Tracker
    </header>
  );
}

export default Header;
