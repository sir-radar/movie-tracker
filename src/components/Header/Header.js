import React from 'react';
import {Link} from 'react-router-dom';

const Header = () => {
  return (
    <header className="row bg-header p-3 text-light font-weight-bolder">
      <Link
        data-testid="header-link" 
        to='/'
        className="text-white"
      > Movie Tracker
      </Link>
    </header>
  );
}

export default Header;
