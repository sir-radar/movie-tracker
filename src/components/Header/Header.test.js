import React from 'react';
import { MemoryRouter } from "react-router-dom";
import { render, cleanup, fireEvent } from '@testing-library/react';
import Header from './Header';
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer"

afterEach(cleanup)

const renderComponent = () => {
  const div = document.createElement("div");
  return render(<MemoryRouter><Header/></MemoryRouter>, div)
}

describe('Header Component', () => {

  it("renders without crashing", () => {
    renderComponent()
  })

  it("renders the router links", () => {
    const { getByTestId } = renderComponent()
    expect(getByTestId('header-link')).toHaveTextContent("Movie Tracker")

  })

  it("navigates to home page when you click the header link", () => {
    const { getByTestId } = renderComponent()
    fireEvent.click(getByTestId('header-link'));
    expect(location.pathname).toBe('/')
  })

  it("matches snapshot", () => {
    const tree = renderer.create(<MemoryRouter><Header/></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  })

})