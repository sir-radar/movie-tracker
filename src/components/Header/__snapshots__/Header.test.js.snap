// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<header
  className="row bg-header p-3 text-light font-weight-bolder"
>
  <a
    className="text-white"
    data-testid="header-link"
    href="/"
    onClick={[Function]}
  >
     Movie Tracker
  </a>
</header>
`;
